src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace NServiceBus
+{
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extentions for <see cref="EndpointConfiguration"/>.
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="endpointName">The endpoint name to be used.</param>
+        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
+        {
+            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
```

Comment:
would it be possible to have some kind of test to make sure we will note in case that config key changes? I think this could be a fairly easy unit test using `context.Settings.EndpointName()` to verify that setting works correctly?